android/app/build.gradle

```diff
@@ -51,7 +52,7 @@ android {
     }
 
     defaultConfig {
-        applicationId "com.react.testapp"
+        applicationId project.ext.react.applicationId
```

Comment:
@arazabishov, @sweggersen: I think we have at least two issues here:
1. Build artifacts are stored in `node_modules/react-native-test-app/`. This can cause issues in a monorepo and the package is hoisted.
2. We need to correct any references to `BuildConfig` and `R`.
```
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/MainActivity.kt: (104, 22): Unresolved reference: R
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/MainActivity.kt: (110, 36): Unresolved reference: R
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/MainActivity.kt: (119, 32): Unresolved reference: R
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/MainActivity.kt: (121, 17): Unresolved reference: R
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/component/ComponentListAdapter.kt: (7, 26): Unresolved reference: R
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/component/ComponentListAdapter.kt: (20, 17): Unresolved reference: R
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/manifest/ManifestProvider.kt: (4, 26): Unresolved reference: R
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/manifest/ManifestProvider.kt: (16, 30): Unresolved reference: R
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/react/TestAppReactNativeHost.kt: (19, 26): Unresolved reference: BuildConfig
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/react/TestAppReactNativeHost.kt: (58, 13): Unresolved reference: BuildConfig
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/react/TestAppReactNativeHost.kt: (81, 13): Unresolved reference: BuildConfig
e: /~/example/node_modules/react-native-test-app/android/app/src/main/java/com/react/testapp/react/TestAppReactNativeHost.kt: (88, 13): Unresolved reference: BuildConfig
```

Any ideas?

Reply:
Can we codegen correct references when creating the project using plop templates? I could imagine `applicationId` being an argument.

Reply:
But what if you change the value of `TEST_APP_IDENTIFIER` after it's been generated. Can we codegen a support class that returns `BuildConfig` and `R` like how autolinking works?

Reply:
Opened #158 for the first issue.

Reply:
Never mind the second issue. I didn't clean up all the Android build artifacts. It works:
```
% adb shell 'pm list packages -f' | grep contoso
package:/data/app/com.contoso.MyTestApp-lGagASJYZMeA8zqIMXmKvA==/base.apk=com.contoso.MyTestApp
```